target/linux/generic/files/drivers/net/phy/rtl8367b.c

```diff
@@ -688,18 +713,24 @@ static int rtl8367b_write_phy_reg(struct rtl8366_smi *smi,
 static int rtl8367b_init_regs(struct rtl8366_smi *smi)
 {
 	const struct rtl8367b_initval *initvals;
+	u32 chip_num;
 	u32 chip_ver;
 	u32 rlvid;
 	int count;
 	int err;
 
 	REG_WR(smi, RTL8367B_RTL_MAGIC_ID_REG, RTL8367B_RTL_MAGIC_ID_VAL);
+	REG_RD(smi, RTL8367B_CHIP_NUMBER_REG, &chip_num);
 	REG_RD(smi, RTL8367B_CHIP_VER_REG, &chip_ver);
 
 	rlvid = (chip_ver >> RTL8367B_CHIP_VER_RLVID_SHIFT) &
 		RTL8367B_CHIP_VER_RLVID_MASK;
 
-	switch (rlvid) {
+	dev_info(smi->parent, "checking chip_num=0x%x ver=0x%x...\n", chip_num, chip_ver);
+	if (chip_num == 0x6367 || chip_num == 0x0597 || chip_num == 0x0276) {
```

Comment:
Where do these chip numbers come from and can we be sure that other chips won't match here?

Reply:
@dengqf6 Can't we use chip_ver for RTL8367S too? Two RTL8367S chips on TL-R600VPN and WSR-2533DHP2 returns ```00a0```.

Reply:
I agree with musashino205. Furthermore I think `case 0` and corresponding initvals don't make sense as both already supported chip versions (0x1000 and 0x1010) return 1 here...
```
RTL8367B_CHIP_VER_RLVID_SHIFT = 12
RTL8367B_CHIP_VER_RLVID_MASK = 0xf
```
`rlvid = (chip_ver >> 12) & 0xf;`
There is no point in calculating this rlvid... Except we use it for our newly supported chip instead:
```
	rlvid = (chip_ver >> RTL8367B_CHIP_VER_RLVID_SHIFT) &
		RTL8367B_CHIP_VER_RLVID_MASK;

	dev_info(smi->parent, "checking chip_num=0x%x ver=0x%x...\n", chip_num, chip_ver);
	switch (rlvid) {
		case 0:
			initvals = rtl8367c_initvals0;
			count = ARRAY_SIZE(rtl8367c_initvals0);
			break;
		case 1:
			initvals = rtl8367r_vb_initvals_1;
			count = ARRAY_SIZE(rtl8367r_vb_initvals_1);
			break;
		default:
			dev_err(smi->parent, "unknown rlvid %u\n", rlvid);
			return -ENODEV;
	}
```

Reply:
And there is also no need for the "unknown rlvid" case I guess as we do a check on the version number at detection stage already. Therefore the lines corresponding to
`dev_info(smi->parent, "checking chip_num=0x%x ver=0x%x...\n", chip_num, chip_ver);` should be moved to the detection function instead (also we don't want to print it on every initialization, do we?).